src/aiofiles/os.py

```diff
@@ -25,34 +26,50 @@
     "wrap",
     "getcwd",
 ]
-if hasattr(os, "link"):
-    __all__ += ["link"]
-if hasattr(os, "sendfile"):
-    __all__ += ["sendfile"]
-if hasattr(os, "statvfs"):
-    __all__ += ["statvfs"]
 
 
-stat = wrap(os.stat)
+access = wrap(os.access)
+
+getcwd = wrap(os.getcwd)
+
+listdir = wrap(os.listdir)
+
+makedirs = wrap(os.makedirs)
+mkdir = wrap(os.mkdir)
+
+readlink = wrap(os.readlink)
+remove = wrap(os.remove)
+removedirs = wrap(os.removedirs)
 rename = wrap(os.rename)
 renames = wrap(os.renames)
 replace = wrap(os.replace)
-remove = wrap(os.remove)
-unlink = wrap(os.unlink)
-mkdir = wrap(os.mkdir)
-makedirs = wrap(os.makedirs)
 rmdir = wrap(os.rmdir)
-removedirs = wrap(os.removedirs)
-symlink = wrap(os.symlink)
-readlink = wrap(os.readlink)
-listdir = wrap(os.listdir)
+
 scandir = wrap(os.scandir)
-access = wrap(os.access)
-getcwd = wrap(os.getcwd)
+stat = wrap(os.stat)
+symlink = wrap(os.symlink)
+
+unlink = wrap(os.unlink)
+
 
 if hasattr(os, "link"):
+    __all__ += ["link"]
     link = wrap(os.link)
 if hasattr(os, "sendfile"):
+    __all__ += ["sendfile"]
     sendfile = wrap(os.sendfile)
 if hasattr(os, "statvfs"):
+    __all__ += ["statvfs"]
     statvfs = wrap(os.statvfs)
+
+
+async def walk(top, topdown=True, onerror=None, followlinks=False):
```

Comment:
Ok, better, but this function isn't really async now. It doesn't handle file system operations in a thread, and so can block the event loop, which we're trying to avoid. 

Instead of a simple for loop, try executing `next()` on the threadpool.

To get you started, maybe try something like

```
source = os.walk(...)

while True:
    f = await loop.run_in_executor(None, next, source)
```